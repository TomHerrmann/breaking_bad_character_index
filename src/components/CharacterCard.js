import React from 'react';

const characterCard = () => {
  return <h2>Name</h2>;
};

export default characterCard;
