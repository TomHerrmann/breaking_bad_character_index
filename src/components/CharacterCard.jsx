import React from 'react';
import removeQuotesAndBraces from '../utils/removeQuotesAndBraces';

const CharacterCard = ({ character }) => {
  const {
    name,
    nickname,
    img,
    portrayed,
    category,
    status,
    occupation,
  } = character;

  return (
    <div className="character-card">
      <div className="img-container">
        <img src={img} alt={`${name} from ${category}`} />
      </div>
      <div className="ul-container">
        <ul>
          <li>
            <strong>Name: </strong>
            {name}
          </li>
          <li>
            <strong>Nickname: </strong>
            {nickname}
          </li>
          <li>
            <strong>Occupation: </strong>
            {removeQuotesAndBraces(occupation)}
          </li>
          <li>
            <strong>Portrayed By: </strong>
            {portrayed}
          </li>
          <li>
            <strong>Status: </strong>
            {status}
          </li>
          <li>
            <strong>First Appeared On: </strong>
            {category}
          </li>
        </ul>
      </div>
    </div>
  );
};

export default CharacterCard;
