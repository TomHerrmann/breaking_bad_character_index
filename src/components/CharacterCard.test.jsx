import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CharacterCard from './CharacterCard';

describe('<CharacterCard />', () => {
  let defaultProps;
  beforeEach(() => {
    defaultProps = {
      character: {
        name: 'thomas',
        nickname: 'tom',
        img: 'profile_pic',
        portrayed: 'himself',
        category: 'person',
        status: 'alive',
        occupation: '{"software engineer","podcast host"}',
      },
    };
  });

  it('should render correctly', () => {
    const component = shallow(<CharacterCard {...defaultProps} />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
