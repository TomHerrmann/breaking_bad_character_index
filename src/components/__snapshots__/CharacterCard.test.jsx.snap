// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CharacterCard /> should render correctly 1`] = `
<div
  className="character-card"
>
  <div
    className="img-container"
  >
    <img
      alt="thomas from person"
      src="profile_pic"
    />
  </div>
  <div
    className="ul-container"
  >
    <ul>
      <li>
        <strong>
          Name: 
        </strong>
        thomas
      </li>
      <li>
        <strong>
          Nickname: 
        </strong>
        tom
      </li>
      <li>
        <strong>
          Occupation: 
        </strong>
        software engineer, podcast host
      </li>
      <li>
        <strong>
          Portrayed By: 
        </strong>
        himself
      </li>
      <li>
        <strong>
          Status: 
        </strong>
        alive
      </li>
      <li>
        <strong>
          First Appeared On: 
        </strong>
        person
      </li>
    </ul>
  </div>
</div>
`;
